sound/soc/sof/intel/hda-stream.c

```diff
@@ -363,12 +363,30 @@ int hda_dsp_stream_hw_params(struct snd_sof_dev *sdev,
 			  sd_offset + SOF_HDA_ADSP_REG_CL_SD_CBL,
 			  hstream->bufsize);
 
+	/*
+	 * Fix for HDA preload issue due to HW restriction
```

Comment:
there are never hw restrictions, only undocumented features :-)

Please change to : recommended hardware programming sequence for HDAudio DMA format.

Reply:
thank, this is a topic our team has discussed how to deal with this type of issue.
This is another solution.